include/ibamr/LevelSetUtilities.h

```diff
@@ -0,0 +1,420 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2017 - 2020 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+/////////////////////////////// INCLUDE GUARD ////////////////////////////////
+
+#ifndef included_IBAMR_LevelSetUtilities
+#define included_IBAMR_LevelSetUtilities
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+
+#include <ibamr/config.h>
+
+#include "ibamr/LSInitStrategy.h"
+
+#include "tbox/DescribedClass.h"
+#include "tbox/Pointer.h"
+#include "tbox/Serializable.h"
+
+#include <limits>
+#include <string>
+
+namespace SAMRAI
+{
+namespace hier
+{
+template <int DIM>
+class PatchHierarchy;
+}
+namespace pdat
+{
+template <int DIM, class TYPE>
+class CellVariable;
+} // namespace pdat
+namespace tbox
+{
+class Database;
+} // namespace tbox
+} // namespace SAMRAI
+
+namespace IBAMR
+{
+class AdvDiffHierarchyIntegrator;
+} // namespace IBAMR
+
+namespace IBTK
+{
+class HierarchyMathOps;
+}
+
+/////////////////////////////// CLASS DEFINITION /////////////////////////////
+
+namespace IBAMR
+{
+/*!
+ * \brief Class LevelSetUtilities provides implementation of some helper functions
+ * to enforce mass/volume conservation of phases.
+ */
+
+class LevelSetUtilities
+{
+public:
+    /*!
+     * \brief A lightweight class to hold the level set variable and the associated hierarchy
```

Comment:
Container has a fairly specific meaning in C++ / STL. Is there a reason that you want to call this a container?

Also, there are is at least one other data item stored (interface half width). Can the description be a little "higher level" --- maybe (just guessing) something like "encapsulate data needed to specify a level set formulation"?

Reply:
Sounds good. Do you have a different name for container here?

Reply:
I'm not sure --- what are you imagining that this class encapsulates?

Reply:
Mainly the LS variable, its hierarchy integrator, and numerical interface width.

Reply:
I mean is it just a "level set function specification" ("level set specification")?